linux/LibSensors.c

```diff
@@ -261,6 +261,43 @@ void LibSensors_getCPUTemperatures(CPUData* cpus, unsigned int existingCPUs, uns
       goto out;
    }
 
+   /* Read sysfs to get HT/SMT cores and correctly distribute the temperature values */
+   if (coreTempCount < activeCPUs) {
+      double oldData[existingCPUs + 1];
+      memcpy(oldData, data, sizeof(oldData));
+
+      int dataCell = 1;
+      for (unsigned int i = 0; i < existingCPUs; i++) {
+         char path[100];
+         sprintf(path, "/sys/devices/system/cpu/cpu%d/topology/thread_siblings_list", i);
+         FILE* file = fopen(path, "r");
```

Comment:
The `fopen` APIs are usually avoided due to their overhead and performance impact. As mentioned above, use `opendir` and `openat` (cf. wrappers in `XUtils.h`) instead.